pkg/extension/sumologicextension/authentication.go

```diff
@@ -0,0 +1,173 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicextension
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/sumologicextension/api"
+	"go.uber.org/zap"
+)
+
+// Authentication is an interface to get collector authentication data
+type Authentication interface {
+	Get() (api.OpenRegisterResponsePayload, error)
+}
+
+// Authenticator is a common structure for both types of getting collector credentials
+// stored in file and registration new collector.
+type Authenticator struct {
+	conf   *Config
+	logger *zap.Logger
+}
```

Comment:
I like this idea but I believe we can change this so that the common struct is unexported (no need to allow others to see what's inside) and change the name of `Authentication` to `Authenticator` (to use the same convention as elsewhere in Golang; `io.Reader`, `io.Writer`, `io.Closer` etc.